src/plugins/data/public/antlr/shared/utils.ts

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { from } from 'rxjs';
+import { distinctUntilChanged, switchMap } from 'rxjs/operators';
+
+export interface IDataSourceRequestHandlerParams {
+  dataSourceId: string;
+  title: string;
+}
+
+export const getRawSuggestionData$ = (
+  connectionsService,
+  dataSourceReuqstHandler: ({
+    dataSourceId,
+    title,
+  }: IDataSourceRequestHandlerParams) => Promise<any>,
+  defaultReuqstHandler: () => any
+) =>
+  connectionsService.getSelectedConnection$().pipe(
+    distinctUntilChanged(),
+    switchMap((connection) => {
```

Comment:
Is this the place to get multi datasource id?

Reply:
yes, and if it's undefined, the request will hit local cluster.

Reply:
Got it. Just wanna make sure the underlying logic can be relied on as a source of truth for the flag of `data_source.enabled`, since my work changed the registration behavior of datasource management plugin. And I've seen that some of the plugin used the datasource management plugin status to determine if multi-datasource is enabled or not.